### PR TITLE
Problem: print_log tasks keep reading after vm end

### PR DESCRIPTION
**Pull request by olethanh**
After the vm proc and the related stdout and stderr were closed the task keep trying to read them, creating a busy loop which wasted cpu.

Solution: end the task when the file descriptor is closed

### CODE REVIEW

File: src/aleph/vm/hypervisors/firecracker/microvm.py [hunk 1]
```diff
@@ -341,25 +342,23 @@ async def print_logs(self):
...
+            if not line:  # EOF, FD is closed nothing more will come
+                return
             for queue in self.log_queues:
-                await queue.put(("stdout", stdout))
-            if stdout:
-                print(stdout.decode().strip())
-            else:
-                await asyncio.sleep(0.001)
+                await queue.put(("stdout", line))
+            print(self, line, line.decode().strip())
```
[the remark this answers is not in the document]
Comment (olethanh) [reply src/aleph/vm/hypervisors/firecracker/microvm.py:350]: `line` is a typo, will fix.
`self` is so I can see from which VM this is coming from. (I took the fix from the my qemu implementation where __str__ is overloaded to show the vm_id _

[the remark this answers is not in the document]
Comment (hoh) [reply src/aleph/vm/hypervisors/firecracker/microvm.py:350]: The change in behaviour is fine as long as the result is better :-)

**Comment by hoh** [comment]
Can you rebase this branch on `main` ?

**Review by hoh — APPROVED** [review]

**Comment by olethanh** [comment]
rebased

**Pull request merged**